### ISSUE TITLE
Docs: [no-base-to-string] documented default for ignoredTypeNames is incorrect

### ISSUE THREAD

**Issue by zaewc** [typescript-eslint/typescript-eslint#12463]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [x] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

The `ignoredTypeNames` section of the `no-base-to-string` docs states the wrong default value
Current text:

> The following patterns are considered correct with the default options `{ ignoredTypeNames: ["RegExp"] }`:

However, the rule's actual default includes four entries, not just `RegExp`:

```ts
defaultOptions: [
  {
    checkUnknown: false,
    ignoredTypeNames: ['Error', 'RegExp', 'URL', 'URLSearchParams'],
  },
],
```

The prose should reflect the real default value, for example:

> The following patterns are considered correct with the default options `{ ignoredTypeNames: ["Error", "RegExp", "URL", "URLSearchParams"] }`:

The playground example immediately below currently uses only:

```ts
option='{ "ignoredTypeNames": ["RegExp"] }'
```

Because it appears directly after the “default options” sentence, it reads as if it is showing the default configuration

Updating it to the full default list, or rewording the sentence so that it is clearly just a RegExp-specific example, would remove the confusion

### Affected URL(s)

https://typescript-eslint.io/rules/no-base-to-string/#ignoredtypenames

### Additional Info

The mismatch dates from when `Error`/`URL`/`URLSearchParams` were added to the default `ignoredTypeNames` in the source without the doc's stated default being updated

Separate from the closed #11420, which was about prose wording around RegExp, not the default value

**Comment by StyleShit** [comment]
IMO we can remove this part completely, as we already have the defaults in the first line and the duplication is unnecessary:

> Stringified type names to ignore. Default: ["Error","RegExp","URL","URLSearchParams"].

And maybe add more examples to the snippet, for `Error`, `URL`, and `URLSearchParams`
